### PR TITLE
onagre: init at 1.0.0-alpha.0

### PR DESCRIPTION
**Pull request by jfvillablanca**
###### Description of changes
Add [onagre](https://github.com/oknozor/onagre) application launcher for X and Wayland

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by wegank — CHANGES_REQUESTED** [review]
```
ERROR: The Cargo.lock contains git dependencies
This is currently not supported in the fixed-output derivation fetcher.
Use cargoLock.lockFile / importCargoLock instead.
```

**Comment by jfvillablanca** [comment]
Thanks for the review.
With `importCargoLock.lockFile`, do I just manually copy the lockfile from the original repo or is there an automated/nix-way to generate it from github?

**Comment by wegank** [comment]
> With `importCargoLock.lockFile`, do I just manually copy the lockfile from the original repo or is there an automated/nix-way to generate it from github?

It may be possible to patch the lock file directly without using the methods in the prompt. Otherwise, importing custom Cargo locks is purely manual.

File: pkgs/applications/misc/onagre/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  cargoSha256 = "sha256-IOhAGrAiT2mnScNP7k7XK9CETUr6BjGdQVdEUvTYQT4=";
+
+  nativeBuildInputs = [ cmake pkgconf ];
+  buildInputs = [ freetype expat ];
+
+  meta = with lib; {
+    description = "A general purpose application launcher for X and wayland inspired by rofi/wofi and alfred";
+    homepage = "https://github.com/oknozor/onagre";
+    license = licenses.mit;
+    maintainers = [ maintainers.jfvillablanca ];
```
Comment (wegank) [inline pkgs/applications/misc/onagre/default.nix:37]:
```suggestion
    maintainers = [ maintainers.jfvillablanca ];
    platforms = platforms.linux;
```

**Comment by wegank** [comment]
Please squash the commits for onagre. Otherwise, LGTM.

**Review by wegank — APPROVED** [review]

**Pull request merged**